### PR TITLE
 feat(libs): add SVG rendering support

### PR DESCRIPTION
**Pull request by onecoolx**
Add a standalone svg parsing and rendering library, independent of thorvg and using vector api for rendering.

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` (`astyle v3.4.12` needs to installed by running `cd scripts; ./install_astyle.sh`) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

hi all,
    We had implemented an SVG parsing and rendering library. It does not depends on other libs, uses LVGL's Vector API to render svg images, and can support GPU by Vector API backend. Currently, it support the image and animation attribute parsing in the standard SVG tiny 1.2. In the future, We also want to add some other commonly usely SVG extensions, such as image filters, interactive operations, SVG Font, etc. Do all think this is a good idea?
    cc @kisvegabor  

[https://github.com/lvgl/lvgl/issues/6022](https://github.com/lvgl/lvgl/issues/6022)

### CODE REVIEW

**Comment by onecoolx** [comment]
cc @XuNeo @FASTSHIFT @bjsylvia @terry0012 @xiaoxiang781216

**Comment by progerstar** [comment]
Could you show an usage example?

**Comment by onecoolx** [comment]
> Could you show an usage example?

you can see the ”test_draw_svg.c” file

**Comment by onecoolx** [comment]
https://github.com/lvgl/lvgl/issues/6022

**Comment by kisvegabor** [comment]
It's definitely a good idea! It looks like a very complete and complex implementation. 

A few general comments:
- lv_tree
  - Can you use `lv_array` to eliminate some redundancy?
  - Is `class`es are really required here? 
  - I was thinking about using it in `lv_obj_tree.c` as well, but realized that [lv_tree_node_t](https://github.com/lvgl/lvgl/pull/6845/files#diff-ae64225a7f2084ad53f925ba25e7f36e1318b85e2786e684ede666379e195416R43-R47) uses more RAM than our current array-only approach (it has more feature in exchange for the higher RAM usage)
- SVG
  - `lv_svg_attr_type_t` and the other in `lv_svg.h` should be hidden from the users. Consider creating an `lv_svg_private.h` file.
  - We should have a widget-like wrapper. Similar to `lv_lottie` to make the usage simpler. Ideally there should be an image decoder and handled by `lv_image`.
- Others:
  - Please check out this [API comment guide](https://github.com/lvgl/lvgl/blob/master/docs/CODING_STYLE.rst#comments) 
  - Please add a few examples too  
  - Where can we set the width and height when the vectors are rendered?

**Comment by onecoolx** [comment]
> It's definitely a good idea! It looks like a very complete and complex implementation.
> 
> A few general comments:
> 

OK, let me explain them separately.

> * lv_tree
>   
>   * Can you use `lv_array` to eliminate some redundancy?
>   * Is `class`es are really required here?

A common tree structure is introduced to simplify the code. SVG follows the XML specification. Parsing SVG requires generating a DOM tree and then traversing the tree structure to get the correct drawing list.

I also realized that tree structures have many other uses, especially in GUI systems, where UI components are usually organized into tree structures, like lvgl 's objs, or event dispatching systems, etc. `lv_tree_t` is a general implementation of tree structures that supports pre_order and post_order traversal methods. The code can be optimized and enhanced to meet various needs.

>   * I was thinking about using it in `lv_obj_tree.c` as well, but realized that [lv_tree_node_t](https://github.com/lvgl/lvgl/pull/6845/files#diff-ae64225a7f2084ad53f925ba25e7f36e1318b85e2786e684ede666379e195416R43-R47) uses more RAM than our current array-only approach (it has more feature in exchange for the higher RAM usage)

We can add some configuration and params to control the memory allocation strategy.


> * SVG
>   
>   * `lv_svg_attr_type_t` and the other in `lv_svg.h` should be hidden from the users. Consider creating an `lv_svg_private.h` file.
In my original conception, `lv_svg.h` is the public SVG API, which contains the data structures and definitions needed to generate the SVG DOM tree. Users need to access the interfaces in this file to access and manipulate the DOM tree, for example, in SVG Animation, an animation node needs to access the properties of its parent node, etc. `lv_svg_token.h` and `lv_svg_parser.h` are private, they are SVG's tokenizer and parser. `lv_svg_render.h` provides the function of drawing SVG images, which is also a public API.

>   * We should have a widget-like wrapper. Similar to `lv_lottie` to make the usage simpler. Ideally there should be an image decoder and handled by `lv_image`.

Yes, this is what we are going to do next. We are going to encapsulate a widget to play SVG animations and images. We will also implement a decoder for SVG format files.

> * Others:
>   
>   * Please check out this [API comment guide](https://github.com/lvgl/lvgl/blob/master/docs/CODING_STYLE.rst#comments)
>   * Please add a few examples too

Yes， I will add an new demo for it.

>   * Where can we set the width and height when the vectors are rendered?

It can be the height and width of the target canvas or widget.

**Comment by kisvegabor** [comment]
Sounds good! I'll review it again when the examples and docs are added. Thank you for the great work! :+1:

**Comment by xinsuinizhuan** [comment]
> > Could you show an usage example?
> 
> you can see the ”test_draw_svg.c” file

where is the test_draw_svg.c file?

**Comment by onecoolx** [comment]
> > > Could you show an usage example?
> > 
> > 
> > you can see the ”test_draw_svg.c” file
> 
> where is the test_draw_svg.c file?

[tests/src/test_cases/test_svg.c](https://github.com/lvgl/lvgl/pull/6845/files#diff-2e2b84e1eaeca1179ac444fc641030c28fb8d3e16ad1a81459888b76f0a2e891)

**Comment by onecoolx** [comment]
add an SVG render demo

**Comment by xinsuinizhuan** [comment]
when to add to the master? the newest code could not have?

**Comment by xinsuinizhuan** [comment]
how to add the svg to a led or meter?

File: demos/svg/lv_demo_svg.c [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+/*********************
+ *      INCLUDES
+ *********************/
+#include "lv_demo_svg.h"
+
+#if LV_USE_DEMO_SVG
+
+extern size_t svg_len;
+extern char svg_data[];
```
Comment (liamHowatt) [inline demos/svg/lv_demo_svg.c]: Should these symbols be namespaced a bit more? Maybe `img_demo_svg_tiger_len` and `img_demo_svg_tiger_data` to match some of the other demo assets?

File: demos/svg/lv_demo_svg.c [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+ *   STATIC FUNCTIONS
+ **********************/
+static void event_cb(lv_event_t * e)
+{
+    lv_layer_t * layer = lv_event_get_layer(e);
+    lv_vector_dsc_t * dsc = lv_vector_dsc_create(layer);
+
+    lv_area_t rect = {0, 0, WIDTH, HEIGHT};
+    lv_vector_dsc_set_fill_color(dsc, lv_color_white());
+    lv_vector_clear_area(dsc, &rect); // clear screen
```
Comment (liamHowatt) [inline demos/svg/lv_demo_svg.c]:
```suggestion
    lv_vector_clear_area(dsc, &rect); /*clear screen*/
```

**Review by liamHowatt — COMMENTED** [review]
Impressive work! I will continue reviewing soon. Looking good in general.

File: demos/svg/lv_demo_svg.c (hunk 1 above)
Comment (onecoolx) [reply demos/svg/lv_demo_svg.c]: done

File: demos/svg/lv_demo_svg.c (hunk 2 above)
Comment (onecoolx) [reply demos/svg/lv_demo_svg.c]: done

File: demos/svg/lv_demo_svg.c [hunk 3]
```diff
@@ -0,0 +1,80 @@
+/**
+ * @file lv_demo_svg.c
```
Comment (kisvegabor) [inline demos/svg/lv_demo_svg.c]: Please add it as an example instead.

File: docs/libs/svg.rst [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+Usage
+-----
+
+Enable :c:macro:`LV_USE_SVG` in ``lv_conf.h``.
+
+See the examples below.
+
+If you need support SVG animation attribute parsing, 
+you can enable :c:macro:`LV_USE_SVG_ANIMATION` in ``lv_conf.h``.
+
```
Comment (kisvegabor) [inline docs/libs/svg.rst:20]: Please ass some more detail about the main API functions, how to use them, etc.

**Review by kisvegabor — COMMENTED** [review]
Amazing! Special thanks for the many tests!  Just a few minor comments.

The only main thing still missing for me is the easy-to-use SVG image widget or an SVG image decoder. @XuNeo can you join the discussion regarding the SVG image decoder?

File: demos/svg/lv_demo_svg.c (hunk 3 above)
Comment (onecoolx) [reply demos/svg/lv_demo_svg.c]: The example is here [lv_example_svg_1.c](https://github.com/lvgl/lvgl/pull/6845/files#diff-c4ee2b6a8bbab84f9d61b9247509f95fae1d626aeb25a40681104e37e08f3498)

**Comment by XuNeo** [comment]
I imagine the newly added SVG lib should provide interface to parse SVG file to generate a data structure that can be rendered any time.

The lv_image_header_t->flags should use bit `LV_IMAGE_FLAGS_VECTORS ` to indicate the image is a vector image, and void *data is the actual generated data structure.

https://github.com/lvgl/lvgl/pull/5970
The above PR is one attempt to allow such data structure independent of drawing information.

The vector API is originally design and used by @onecoolx , please suggest the correct way to add SVG to image decoder.

**Comment by onecoolx** [comment]
> I imagine the newly added SVG lib should provide interface to parse SVG file to generate a data structure that can be rendered any time.
> 
> The lv_image_header_t->flags should use bit `LV_IMAGE_FLAGS_VECTORS ` to indicate the image is a vector image, and void *data is the actual generated data structure.
> 
> #5970 The above PR is one attempt to allow such data structure independent of drawing information.
> 
> The vector API is originally design and used by @onecoolx , please suggest the correct way to add SVG to image decoder.



> I imagine the newly added SVG lib should provide interface to parse SVG file to generate a data structure that can be rendered any time.
> 
> The lv_image_header_t->flags should use bit `LV_IMAGE_FLAGS_VECTORS ` to indicate the image is a vector image, and void *data is the actual generated data structure.
> 
> #5970 The above PR is one attempt to allow such data structure independent of drawing information.
> 
> The vector API is originally design and used by @onecoolx , please suggest the correct way to add SVG to image decoder.

Thanks to @XuNeo, I think for static images, we don't need to keep the DOM tree. `*data` can save the pointer of `lv_svg_render_obj_t` generated by SVG parsing. This pointer is a drawing list. When drawing is needed, it can be drawn using the vector API. The only thing you need to solve is to call `lv_svg_render_delete` to release it. Maybe we also need to estimate the size of the diswing list so that the cache system can use it?

File: docs/libs/svg.rst [hunk 5]
```diff
@@ -25,8 +25,25 @@ Example
 
 .. include:: ../examples/libs/svg/index.rst
```
Comment (kisvegabor) [inline docs/libs/svg.rst:26]: Please move this below your example.

File: demos/svg/lv_demo_svg.c [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+ *
+ */
+
+/*********************
+ *      INCLUDES
+ *********************/
+#include "lv_demo_svg.h"
+
+#if LV_USE_DEMO_SVG
```
Comment (kisvegabor) [inline demos/svg/lv_demo_svg.c]: I meant this demo is not required. We usually put mode complex UIs into the demos folder. It's enough to keep it as an example.

Comment (onecoolx) [reply demos/svg/lv_demo_svg.c]: done

**Comment by onecoolx** [comment]
> The image decoder as you have described makes sense to me. Could you add it still in this PR?

I think this is not a good idea. We shouldn't include too many different things in one PR, it will clutter the commit log. I will open a new PR to add the decoder part.

File: docs/libs/svg.rst (hunk 5 above)
Comment (onecoolx) [reply docs/libs/svg.rst:26]: Let me confirm, do you want to put the following code into the example document? Or move the whole "svg.rst" to the example directory?

**Comment by xinsuinizhuan** [comment]
when SVG merger to master?

**Comment by onecoolx** [comment]
> when SVG merger to master?

In reviewing

**Comment by kisvegabor** [comment]
@onecoolx I've tested it but `lv_example_svg_1` crashed with:
```
=================================================================
[1m[31m==37586==ERROR: AddressSanitizer: global-buffer-overflow on address 0x5555560c29ab at pc 0x555555854fdf bp 0x7fffffffced0 sp 0x7fffffffcec0
[1m[0m[1m[34mREAD of size 1 at 0x5555560c29ab thread T0[1m[0m
    #0 0x555555854fde in _lv_svg_tokenizer ../lvgl/src/libs/svg/lv_svg_token.c:369
    #1 0x5555558363d2 in lv_svg_load_data ../lvgl/src/libs/svg/lv_svg.c:87
    #2 0x555555a1dd0b in event_cb ../lvgl/examples/libs/svg/lv_example_svg_1.c:14
    #3 0x5555556bb5d5 in lv_event_send ../lvgl/src/misc/lv_event.c:98
    #4 0x5555559d098d in event_send_core ../lvgl/src/core/lv_obj_event.c:367
    #5 0x5555559cf3da in lv_obj_send_event ../lvgl/src/core/lv_obj_event.c:67
    #6 0x5555559f4dcd in lv_obj_redraw ../lvgl/src/core/lv_refr.c:116
    #7 0x5555559fb59b in refr_obj ../lvgl/src/core/lv_refr.c:1081
    #8 0x5555559fab8c in refr_obj_and_children ../lvgl/src/core/lv_refr.c:869
    #9 0x5555559fa708 in refr_configured_layer ../lvgl/src/core/lv_refr.c:803
    #10 0x5555559f9dd7 in refr_area ../lvgl/src/core/lv_refr.c:724
    #11 0x5555559f8c8f in refr_invalid_areas ../lvgl/src/core/lv_refr.c:626
    #12 0x5555559f6bf4 in lv_display_refr_timer ../lvgl/src/core/lv_refr.c:400
    #13 0x5555556dc4f1 in lv_timer_exec ../lvgl/src/misc/lv_timer.c:325
    #14 0x5555556db341 in lv_timer_handler ../lvgl/src/misc/lv_timer.c:107
    #15 0x555555a4a9d3 in main ../main.c:129
    #16 0x7ffff6d32d8f in __libc_start_call_main ../sysdeps/nptl/libc_start_call_main.h:58
    #17 0x7ffff6d32e3f in __libc_start_main_impl ../csu/libc-start.c:392
    #18 0x5555555e9a84 in _start (/home/kisvegabor/projects/lvgl/eclipse-workspace/lv_port_pc_eclipse/Debug/lv_port_pc_eclipse+0x95a84)

[1m[32m0x5555560c29ab is located 0 bytes to the right of global variable 'svg_data' defined in '../lvgl/examples/libs/svg/lv_example_svg_1.c:10:17' (0x5555560c2920) of size 139
[1m[0m  'svg_data' is ascii string '<svg width="12cm" height="4cm" viewBox="0 0 1200 400"><circle cx="600" cy="200" r="100" fill="red" stroke="blue" stroke-width="10"/></svg>'
SUMMARY: AddressSanitizer: global-buffer-overflow ../lvgl/src/libs/svg/lv_svg_token.c:369 in _lv_svg_tokenizer
Shadow bytes around the buggy address:
  0x0aab2ac104e0: [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[0m00[1m[0m [1m[0m00[1m[0m [1m[0m00[1m[0m [1m[0m00[1m[0m
  0x0aab2ac104f0: [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m
  0x0aab2ac10500: [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m
  0x0aab2ac10510: [1m[0m00[1m[0m [1m[0m00[1m[0m [1m[0m00[1m[0m [1m[0m00[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m
  0x0aab2ac10520: [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[0m00[1m[0m [1m[0m00[1m[0m [1m[0m00[1m[0m [1m[0m00[1m[0m [1m[0m00[1m[0m [1m[0m00[1m[0m [1m[0m00[1m[0m [1m[0m00[1m[0m [1m[0m00[1m[0m [1m[0m00[1m[0m [1m[0m00[1m[0m [1m[0m00[1m[0m
=>0x0aab2ac10530: [1m[0m00[1m[0m [1m[0m00[1m[0m [1m[0m00[1m[0m [1m[0m00[1m[0m [1m[0m00[1m[0m[[1m[0m03[1m[0m][1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[0m00[1m[0m [1m[0m00[1m[0m [1m[0m00[1m[0m [1m[0m00[1m[0m
  0x0aab2ac10540: [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m
  0x0aab2ac10550: [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m
  0x0aab2ac10560: [1m[0m00[1m[0m [1m[0m00[1m[0m [1m[0m00[1m[0m [1m[0m00[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m
  0x0aab2ac10570: [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m
  0x0aab2ac10580: [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[31mf9[1m[0m [1m[0m00[1m[0m [1m[0m00[1m[0m [1m[0m00[1m[0m [1m[0m00[1m[0m
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           [1m[0m00[1m[0m
  Partially addressable: [1m[0m01[1m[0m [1m[0m02[1m[0m [1m[0m03[1m[0m [1m[0m04[1m[0m [1m[0m05[1m[0m [1m[0m06[1m[0m [1m[0m07[1m[0m 
  Heap left redzone:       [1m[31mfa[1m[0m
  Freed heap region:       [1m[35mfd[1m[0m
  Stack left redzone:      [1m[31mf1[1m[0m
  Stack mid redzone:       [1m[31mf2[1m[0m
  Stack right redzone:     [1m[31mf3[1m[0m
  Stack after return:      [1m[35mf5[1m[0m
  Stack use after scope:   [1m[35mf8[1m[0m
  Global redzone:          [1m[31mf9[1m[0m
  Global init order:       [1m[36mf6[1m[0m
  Poisoned by user:        [1m[34mf7[1m[0m
  Container overflow:      [1m[34mfc[1m[0m
  Array cookie:            [1m[31mac[1m[0m
  Intra object redzone:    [1m[33mbb[1m[0m
  ASan internal:           [1m[33mfe[1m[0m
  Left alloca redzone:     [1m[34mca[1m[0m
  Right alloca redzone:    [1m[34mcb[1m[0m
  Shadow gap:              [1m[0mcc[1m[0m
==37586==ABORTING

```

**Comment by kisvegabor** [comment]
It seems that after 10 cycles in `_lv_svg_tokenizer` `state.cur` will point to the string null terminator, but it's still incremented one more time. 
image

**Comment by onecoolx** [comment]
> It seems that after 10 cycles in `_lv_svg_tokenizer` `state.cur` will point to the string null terminator, but it's still incremented one more time. image

I think this is just a warning from asan. Actually, due to `while(state.cur <= state.end)`, the pointer beyond the buffer will not be used. I modified the code and added a finish label just to make asan happy.

**Comment by kisvegabor** [comment]
Thanks! Works locally, but the CI timeouts. Maybe fixed by https://github.com/lvgl/lvgl/pull/6959?

**Comment by onecoolx** [comment]
> Thanks! Works locally, but the CI timeouts. Maybe fixed by #6959?

done

**Comment by kisvegabor** [comment]
I've just merged https://github.com/lvgl/lvgl/pull/7052 which caused a conflict. :slightly_frowning_face: 

Other than that LGTM assuming that an SVG image decoder will be added too.

**Comment by onecoolx** [comment]
> I've just merged #7052 which caused a conflict. 🙁
> 
> Other than that LGTM assuming that an SVG image decoder will be added too.

The conflict has been resolved

**Review by kisvegabor — APPROVED** [review]

**Review by FASTSHIFT — APPROVED** [review]

**Comment by kisvegabor** [comment]
Yes it will be part of v9.3.

**Pull request merged**